src/storage/restrict.rs

```diff
@@ -165,7 +166,7 @@ where
     }
 }
 
-impl<'rf, 'st: 'rf, B, T, R, RT> Join for &'rf RestrictedStorage<'rf, 'st, B, T, R, RT>
+impl<'rf, 'st: 'rf, B, T, R> Join for &'rf RestrictedStorage<'rf, 'st, B, T, R, ImmutableParallelRestriction>
```

Comment:
Shouldn't this be `ImmutableAliasing` or am I understanding it wrong?

Reply:
Actually I am not sure what's wrong with allowing it for `MutableParallelRestriction` too...
The `ParJoin` is implemented for mutable reference of it so you cannot use `join` to break anything as `Entry`s are tied to `rf` lifetime.
(Not that it's that useful for it)

Reply:
Or is this what you mean be better inference? If that's the case it's fine but I would like to have comment here indicating it.

Reply:
In all technicality yes it can be any restriction, but it is all but useless except with `ImmutableParallelRestriction` and confuses Rust since it defaults to immutable referenced implementations.